### PR TITLE
fix(grid): restore trailing scroll emission so top rows aren't lost after fast scroll

### LINKED ISSUE #17287: Grids: top rows become invisible after fast scrolling to top with inertia
### Description

When scrolling quickly up and down in an igx-grid/ igx-tree-grid, the top rows can occasionally disappear. After a fast momentum/inertia scroll back to the top, the grid visually starts around row 10–15 even though the scroll container reports scrollTop as 0 or near 0.
This appears to be a regression. The same code and configuration works correctly in @infragistics/igniteui-angular@21.0.13.

### Framework

Angular

### Angular Version

21.x

### Ignite UI for Angular Version

21.2.x

### Component / Area

Grid

### Browser

Chrome

### Operating System

_No response_

### Command Not Working

```shell

```

### Steps to Reproduce

1. Open [this](https://stackblitz.com/edit/grid-scroll-sample-xzmhxocs?file=src%2Fapp%2Ftree-grid-primaryforeignkey-sample%2Fdata.ts) sample
2. Scroll the grid to the bottom by holding the scrollbar.
3. Now scroll back to top by using the momentum of the mouse scroll wheel or the touch pad
4. Observe the top records - some are out of view.
5. Position gets recovered after slightly scrolling down&up afterwards

5. Do the same in [this](https://stackblitz.com/edit/grid-scroll-sample-21-0-13?file=src%2Fapp%2Fapp.config.ts) identical sample against version 21.0.13

### Actual Result

Image

### Expected Result

In 21.0.13:

Image

### Reproduction URL

https://stackblitz.com/edit/grid-scroll-sample-xzmhxocs?file=src%2Fapp%2Ftree-grid-primaryforeignkey-sample%2Fdata.ts

### Attachments

_No response_

### LINKED ISSUE #17432: Grid leaves a vertical gap when navigating through a large number of rows
### Description

When navigating vertically through a Grid with a large number of rows, a gap appears in the vertical scroll area.

The issue can be reproduced in the `gridFinJs` sample by navigating from the first row to the last row using `Ctrl + Arrow Down`. After reaching the last row, the vertical scroll position is calculated incorrectly and empty space remains below the rendered rows.

The issue is reproducible in both zone-based and zoneless environments.

### Framework

Angular

### Angular Version

22.0

### Ignite UI for Angular Version

22.1.x

### Component / Area

Grid

### Browser

Chrome

### Operating System

*No response*

### Command Not Working

```shell
```

### Steps to Reproduce

1. Open the `gridFinJs` sample.
2. Focus a cell in the first row.
3. Navigate to the last row using `Ctrl + Arrow Down`.

### Actual Result

After navigating to the last row, a large gap appears below the rendered rows in the vertical scroll area.

### Expected Result

The Grid should navigate to the last row without leaving empty space in the vertical scroll area.

### Reproduction URL

*No response*

### Attachments

Image

### LINKED ISSUE #17464: igxGrid — Rows randomly clipped/cut off during vertical scroll when horizontal scrollbar is present
**Description:**

When an `igx-grid` has enough columns to cause horizontal overflow (horizontal scrollbar visible), rows get randomly clipped during vertical scrolling. Some rows render at partial height or are completely invisible until the user scrolls past them and back.

**Version:** `@infragistics/igniteui-angular` 21.2.7

**Angular version:** 21.x

**Browser:** Chrome (latest), also reproducible in Firefox/Safari

---

**Steps to reproduce:**

1. Create a bare `igx-grid` with `height="100%"`, `width="100%"`, and 100 rows of data
2. Add 6+ columns with default widths (enough for total column width to exceed the viewport, triggering horizontal scroll)
3. Scroll down vertically
4. Observe: rows are randomly clipped — some show at partial height, some are skipped entirely

**Key condition:** The bug only occurs when a horizontal scrollbar is present. Reducing columns so everything fits within the viewport (no horizontal scroll) eliminates the issue entirely.

---

**Minimal reproduction:**

```typescript
@Component({
  selector: 'app-grid-repro',
  standalone: true,
  imports: [IgxGridComponent, IgxColumnComponent],
  template: `
    <igx-grid [data]="data" [autoGenerate]="false" primaryKey="id" height="100%" width="100%">
      <igx-column field="id" header="ID"></igx-column>
      <igx-column field="col1" header="Column 1"></igx-column>
      <igx-column field="col2" header="Column 2"></igx-column>
      <igx-column field="col3" header="Column 3"></igx-column>
      <igx-column field="col4" header="Column 4"></igx-column>
      <igx-column field="col5" header="Column 5"></igx-column>
      <igx-column field="col6" header="Column 6"></igx-column>
      <igx-column field="col7" header="Column 7"></igx-column>
      <igx-column field="col8" header="Column 8"></igx-column>
      <igx-column field="col9" header="Column 9"></igx-column>
    </igx-column>
  `,
  styles: [`:host { display: block; height: 100%; width: 100%; }`],
})
export class GridReproComponent {
  data = Array.from({ length: 100 }, (_, i) => ({
    id: i + 1,
    col1: `Value ${i}`, col2: `Data ${i}`, col3: `Item ${i}`,
    col4: `Record ${i}`, col5: `Entry ${i}`, col6: `Field ${i}`,
    col7: `Cell ${i}`, col8: `Row ${i}`, col9: `Col ${i}`,
  }));
}
```

Host the component in a container with a defined height (e.g., router outlet with `height: 100%`).

---

**Expected behavior:**
All rows render at full consistent height during vertical scrolling regardless of horizontal scroll state.

**Actual behavior:**
Rows are randomly clipped/cut off. The issue is inconsistent — different rows may be affected on each scroll interaction.

---

**What I've ruled out:**
- Not caused by custom directives, wrappers, or cell templates
- Not fixed by setting explicit `[rowHeight]`
- Not related to `autoGenerate`, autosizing, or paginator
- Occurs with a completely bare `igx-grid` + `[data]` binding
- Only trigger: total column width > viewport width (horizontal scrollbar visible)

---

**Possibly related issues:**
- #4931 — "Rendered and predefined heights of the rows is different which leads to all kinds of desyncs"
- #3950 — "The grid miscalculates its height"
- #7602 — Bottom row not visible when grid-footer is used

---

### LINKED ISSUE #17484: IgxGrid with a lot of columns and rows does not have consitant scrolling
### Description

Having a grid with a lot of rows and columns has broken scrolling behavior when scrolling fast.

### Framework

Angular

### Angular Version

latest

### Ignite UI for Angular Version

latest

### Component / Area

Grid

### Browser

Chrome

### Operating System

_No response_

### Command Not Working

```shell

```

### Steps to Reproduce

1. Open grid performance dev sample
2. Scroll with the scrollbar thumb down and up fast

### Actual Result

The scrollbar thumb position goes to the top but does not show the first row. The grid state is desynced. This is happening because of the scroll throttle.

https://github.com/user-attachments/assets/d8e66674-c2cb-410f-9a9b-f99ea6895e65

### Expected Result

The grid should scroll correctly to the top.

### Reproduction URL

_No response_

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by Zneeky**
Closes #17287
Closes #17484 
Closes #17432
Closes #17464 

## Description
Fixes a regression where the top rows of `igx-grid` / `igx-tree-grid` /
`igx-hierarchical-grid` could become invisible after a fast momentum/inertia
scroll back to the top. The scrollbar reports `scrollTop ≈ 0`, but the
virtualized rows stay frozen at an intermediate `startIndex` (~row 10–15).

The vertical-scroll `throttle` on `scrollNotify` was passing no config, so it
used RxJS's default `{ leading: true, trailing: false }`. Restoring the explicit
`{ leading: true, trailing: true }` config ensures the final settle position of
a scroll is always processed.

## Motivation / Context
The throttle originally used
`throttleTime(THROTTLE_TIME, animationFrameScheduler, { leading: false, trailing: true })`.
When it was changed to a data-size–dependent `throttle(durationSelector)` (in
"feat(grids): Apply different throttle based on the amount of data in display"),
the config object was dropped and the operator fell back to the RxJS default
`{ leading: true, trailing: false }`.

Without the trailing edge, the last scroll events of a momentum flick — including
the `scrollTop ≈ 0` settle at the top — fall inside a throttle window (up to 60 ms
for large grids) and are discarded. `verticalScrollContainer.onScroll` never runs
for the final position, so virtualization stays at the last *leading* `startIndex`
while the DOM scrollbar settles at the top → the top rows appear missing.
`trailing: true` restores the guarantee that the final resting scroll position is
always handled. Independent of zone/zoneless change detection.

### Type of Change (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Breaking change
 - [ ] Refactoring (no functional changes)
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD
 - [ ] Tests
 - [ ] Changelog
 - [ ] Skills/Agents

### Component(s) / Area(s) Affected:
Grid, Tree Grid, Hierarchical Grid (shared `IgxGridBaseDirective` vertical
scroll handling / virtualization)

## How Has This Been Tested?
 - [ ] Unit tests
 - [x] Manual testing
 - [ ] Automated e2e tests

Manual: Grid performance sample with 1000 rows — repeatedly wheel-scroll up/down
with momentum and confirm the grid always snaps back to row 0 at the top
(scrollbar at top ⇔ first row visible), where it previously froze around row
10–15.

### Test Configuration:
 - **Angular version**: <fill in>
 - **Browser(s)**: <fill in>
 - **OS**: <fill in>

## Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them
 - [ ] Accessibility (ARIA, keyboard navigation, focus management) has been verified

### CODE REVIEW

File: projects/igniteui-angular/grids/grid/src/grid.component.spec.ts [hunk 1]
```diff
@@ -2161,6 +2161,38 @@ describe('IgxGrid Component Tests #grid', () => {
 
                 expect(grid.headerContainer.state.startIndex).toBeGreaterThan(0);
             });
+
+            it('should settle at the top row when a fast momentum scroll ends at scrollTop 0 (throttle trailing edge)', async () => {
```
Comment (viktorkombov) [inline projects/igniteui-angular/grids/grid/src/grid.component.spec.ts]: Is there a reason this test is placed under `Zoneless rendering regressions`? The changed logic is in the RxJS throttle configuration for `scrollNotify`, and the PR description states that the issue is independent of zone/zoneless change detection. Would it be more appropriate to place it under the general virtualization tests?

Comment (viktorkombov) [inline projects/igniteui-angular/grids/grid/src/grid.component.spec.ts]: `ZonelessTallGridComponent`, used by this test, auto-generates only three columns (`ID`, `Name`, and `Value`) inside a `600px`-wide grid, so it does not render a horizontal scrollbar.

In my manual comparison against `master`, I could not reproduce the UI issue with only these three columns. It became reproducible at both vertical boundaries after adding enough columns to create horizontal overflow.

Because this test emits directly through `grid.scrollNotify`, it verifies the throttle configuration but bypasses the actual DOM scroll conditions. Could we add regression coverage using a fixture that guarantees horizontal overflow and asserts that it is present?

**Review by viktorkombov — CHANGES_REQUESTED** [review]

**Comment by viktorkombov** [comment]
I also verified that this change fixes the bottom-scroll issue reported in #17464. However, the current regression test covers only the top boundary. If we want this PR to close #17464, could we add the corresponding bottom-boundary case and then include `Closes #17464` in the description?

**Review by skrustev — APPROVED** [review]

**Review by viktorkombov — APPROVED** [review]
